packages/client/hmi-client/src/services/dataset.ts

```diff
@@ -228,6 +230,92 @@ export const saveDataset = async (
 	return createDatasetFromSimulationResult(projectId, simulationId, datasetName);
 };
 
+/**
+ * This is a client side function to create a CsvAsset similar to the getCsv function from DatasetResource.java
+ * The reason we reimplement this client side is because runResults already has all of the data we need, except
+ * it's not in the correct format necessary to be consumed by the TeraDatasetDatatable component.
+ * @param runResults
+ * @param runId
+ * @returns CsvAsset object with the data from the runResults
+ */
+const createCsvAssetFromRunResults = (runResults: RunResults, runId?: string): CsvAsset | null => {
+	const runResult: RunResults = _.cloneDeep(runResults);
+	const runIdList = runId ? [runId] : Object.keys(runResult);
+	if (runIdList.length === 0) return null;
+
+	const csvColHeaders = Object.keys(runResult[runIdList[0]][0]);
+	let csvData: CsvAsset = {
+		headers: csvColHeaders,
+		data: [],
+		csv: [csvColHeaders],
+		rowCount: 0,
+		stats: []
+	};
+
+	const csvColumns: { [key: string]: number[] } = {};
+
+	runIdList.forEach((id) => {
+		csvData = {
+			...csvData,
+			data: [...csvData.data, ...(runResult[id] as any)],
+			rowCount: csvData.rowCount + runResult[id].length
+		};
+		runResult[id].forEach((row) => {
+			const rowValues = Object.values(row);
+			csvData.csv.push(rowValues as any);
+
+			csvColHeaders.forEach((header, index) => {
+				if (!csvColumns[header]) {
+					csvColumns[header] = [];
+				}
+				csvColumns[header].push(+rowValues[index]);
+			});
+		});
+	});
+
+	csvColHeaders.forEach((header) => {
+		csvData.stats!.push(getCsvColumnStats(csvColumns[header]));
+	});
+
+	return csvData;
+};
+
+/**
+ * This is a client side implementation of the getStats function from DatasetResource.java
+ * @param csvColumn
+ * @returns CsvColumnStats object
+ */
+const getCsvColumnStats = (csvColumn: number[]): CsvColumnStats => {
```

Comment:
We do already have this functionality in `DatasetResource` as a helper function - It may be more beneficial to just throw an endpoint on that to keep the functionality insync with each other and to offload the more expensive math operations to the server

Reply:
Oh I like that idea, thank you for the suggestion!

Reply:
For visibility: Derek and I had a discussion on slack and decided to leave this as it is for now. We can revisit it if we notice performance issues in the future.